pyClient/zeth/wallet.py

```diff
@@ -6,40 +6,193 @@
 
 from __future__ import annotations
 import zeth.joinsplit as joinsplit
+from zeth.contracts import EncryptedNote, get_merkle_leaf
+from zeth.utils import EtherValue, short_commitment
 from api.util_pb2 import ZethNote
 from nacl.public import PrivateKey, PublicKey  # type: ignore
 from nacl import encoding  # type: ignore
-from os.path import join
-from typing import List, Tuple
-import time
+from os.path import join, basename, exists
+from os import makedirs
+from typing import List, Tuple, Optional, Iterator, Any
+import glob
 import json
 
 
+class ZethNoteDescription:
+    """
+    All secret data about a single ZethNote, including address in the merkle
+    tree and the commit value.
+    """
+    def __init__(self, note: ZethNote, address: int, commitment: bytes):
+        self.note = note
+        self.address = address
+        self.commitment = commitment
+
+    def as_input(self) -> Tuple[int, ZethNote]:
+        """
+        Returns the description in a form suitable for joinsplit.
+        """
+        return (self.address, self.note)
+
+    def to_json(self) -> str:
+        json_dict = {
+            "note": joinsplit.zeth_note_to_json_dict(self.note),
+            "address": str(self.address),
+            "commitment": self.commitment.hex(),
+        }
+        return json.dumps(json_dict, indent=4)
+
+    @staticmethod
+    def from_json(json_str: str) -> ZethNoteDescription:
+        json_dict = json.loads(json_str)
+        return ZethNoteDescription(
+            note=joinsplit.zeth_note_from_json_dict(json_dict["note"]),
+            address=int(json_dict["address"]),
+            commitment=bytes.fromhex(json_dict["commitment"]))
+
+
 class Wallet:
-    def __init__(self, username: str, wallet_dir: str, k_sk_receiver: PrivateKey):
+    """
+    Very simple class to track a list of notes addressed to some owner.  Note
```

Comment:
`Very simple class to track a list of notes addressed to some owner` -> `Very simple class to track the list of notes owned by a user of Zeth`

Reply:
rephrased